lib/modules/manager/pip-compile/common.ts

```diff
@@ -0,0 +1,163 @@
+import is from '@sindresorhus/is';
+import { Command } from 'commander';
+import { split } from 'shlex';
+import upath from 'upath';
+import { logger } from '../../../logger';
+import type { ExecOptions } from '../../../util/exec/types';
+import { ensureCacheDir } from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import type { UpdateArtifactsConfig } from '../types';
+
+export function getPipToolsConstraint(config: UpdateArtifactsConfig): string {
```

Comment:
please do this code move in a separate PR, so we can merge it first. should be non-breaking

Reply:
Done https://github.com/renovatebot/renovate/pull/26873